blueoil/datasets/pascalvoc_base.py

```diff
@@ -129,10 +129,10 @@ def _get_boxes_from_annotation(self, xml_file):
             for e in list(obj):
                 if e.tag == "bndbox":
                     # subtract 1 to make boxes 0-based coordinate.
-                    ymin = int(e.find("ymin").text) - 1
-                    xmin = int(e.find("xmin").text) - 1
-                    ymax = int(e.find("ymax").text) - 1
-                    xmax = int(e.find("xmax").text) - 1
+                    ymin = float(e.find("ymin").text) - 1
+                    xmin = float(e.find("xmin").text) - 1
+                    ymax = float(e.find("ymax").text) - 1
+                    xmax = float(e.find("xmax").text) - 1
```

Comment:
I'm not sure if the coordinates are 1-based also for float values. I feel it's more natural to use 0-based coordinates.

Do we really need to support float values? Is the tool that generates PascalVOC-formatted data aware of 1-based coordinates?

Reply:
> I'm not sure if the coordinates are 1-based also for float values. I feel it's more natural to use 0-based coordinates in that case.

This should be the same as an Integer, as this is a standard for PascalVOC annotation.
I don't have much knowledge about this annotation.
But I think PascalVOC format annotation is  1-based float values. 
Because this PascalVOC dataset loader code in Chainer also process data as 1-based for float values 
https://github.com/chainer/chainercv/blob/7159616642e0be7c5b3ef380b848e16b7e99355b/chainercv/datasets/voc/voc_bbox_dataset.py#L105
In this case, it's better to say that we supported 1-based coordinates for PascalVOC.

> Do we really need to support float values? 

Yes, there is no exact evidence show that the bndbox can be float though, in many PascalVOC document the bndbox value can be float.
- http://host.robots.ox.ac.uk/pascal/VOC/voc2007/guidelines.html (The official website is currently down...)
- https://pjreddie.com/media/files/VOC2007_doc.pdf (Page 13)

> Is the tool that generates PascalVOC-formatted data aware of 1-based coordinates?

Yes, it seems possible to set the offset of the bndbox.

Reply:
Thanks for the evidence. I've understood.